main.js

```diff
@@ -36,11 +37,20 @@ async function main() {
             
             console.log("==> Deleting \"" + ownerOfRepository + "/" + repositoryContainingBranches + "/" + branch + "\" branch")
             
-            await client.git.deleteRef({
-                owner: ownerOfRepository,
-                repo: repositoryContainingBranches,
-                ref: "heads/" + branch
-            })
+            try {
+                await client.git.deleteRef({
+                    owner: ownerOfRepository,
+                    repo: repositoryContainingBranches,
+                    ref: "heads/" + branch
+                })
+            } catch (error) {
+                const shouldFailSoftly = (soft_fail === 'true');
```

Comment:
Can you not have soft_fail as boolean from the start (not as string) in JS?

Reply:
I might have misunderstood something, but it does not seem like it from what is described in https://github.com/actions/toolkit/issues/361

Maybe I just haven't looked into it enough